From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mprotect: use pfn_swap_entry_folio
Date: Thu, 11 Jan 2024 15:24:22 +0000

We only want to know whether the folio is anonymous, so use
pfn_swap_entry_folio() and save a call to compound_head().

Link: https://lkml.kernel.org/r/20240111152429.3374566-4-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mprotect.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/mprotect.c~mprotect-use-pfn_swap_entry_folio
+++ a/mm/mprotect.c
@@ -198,13 +198,13 @@ static long change_pte_range(struct mmu_
 			pte_t newpte;
 
 			if (is_writable_migration_entry(entry)) {
-				struct page *page = pfn_swap_entry_to_page(entry);
+				struct folio *folio = pfn_swap_entry_folio(entry);
 
 				/*
 				 * A protection check is difficult so
 				 * just be safe and disable write
 				 */
-				if (PageAnon(page))
+				if (folio_test_anon(folio))
 					entry = make_readable_exclusive_migration_entry(
 							     swp_offset(entry));
 				else
_
